build-logic/packaging/src/main/kotlin/gradlebuild/instrumentation/tasks/InstrumentedMetadataMergeTask.kt

```diff
@@ -47,25 +52,40 @@ abstract class FindInstrumentedSuperTypesTask : DefaultTask() {
     @get:OutputFile
     abstract val instrumentedSuperTypesHash: RegularFileProperty
 
+    @get:OutputFile
+    abstract val upgradedProperties: RegularFileProperty
+
+    @get:OutputFile
+    abstract val upgradedPropertiesHash: RegularFileProperty
+
     @TaskAction
     fun run() {
         val instrumentedClasses = findInstrumentedClasses()
         if (instrumentedClasses.isEmpty()) {
             instrumentedSuperTypes.asFile.get().toEmptyFile()
             instrumentedSuperTypesHash.asFile.get().toEmptyFile()
-        } else {
-            // Merge and find all transitive super types
-            val superTypes = mergeSuperTypes()
-            // Keep only instrumented super types as we don't need to others for instrumentation
-            val onlyInstrumentedSuperTypes = keepOnlyInstrumentedSuperTypes(superTypes, instrumentedClasses)
-            writeOutput(onlyInstrumentedSuperTypes)
+            upgradedProperties.asFile.get().toEmptyFile()
+            upgradedPropertiesHash.asFile.get().toEmptyFile()
+            return
         }
```

Comment:
❓ Are we getting any benefits by adding this `if` here? It seems both `mergeAndWriteInstrumentedSuperTypes()` and `mergeAndWriteUpgradedProperties()` would do exactly the same at no extra cost without the `if`.

Reply:
For merging super types we do some extra work: we actually collect all super types initially and then we filter just the ones that are actually for instrumented types. So I think fast return makes there. 

Maybe it looks better now, since I splitted `InstrumentedMetadataMergeTask.kt` task in two.